### PR TITLE
AWP-1848:TagsField component

### PR DESCRIPTION
**Pull request by zamanaf**
- Create a component to render options in tags

### CODE REVIEW

**Review by gm-global — COMMENTED** [review]
Can we utilize input component and make this a type of input instead of creating a separate component?

**Comment by zamanaf** [comment]
> Can we utilize input component and make this a type of input instead of creating a separate component?

For this it will need different params I'll label and error in this tagsField component to keep things clean and isolated

File: src/components/tagsField/TagsField.styles.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  border-radius: 0.375rem;
+  border: 1px solid
+    ${({ disabled, error }) =>
+      disabled
+        ? "var(--Neutral-100, #d0d7dd)"
+        : error
+          ? "var(--negative-200, #7f1d1d)"
+          : " var(--neutral-200, #9ca3af)"};
+  background-color: ${({ disabled }) =>
+    disabled ? "var(--Neutral-50, #f9fafb)" : "var(--System-50, #fff)"};
```
Comment (gm-global) [inline src/components/tagsField/TagsField.styles.ts]: Can we move these to the function at the end?

File: src/components/tagsField/TagsField.styles.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  box-shadow: 0px 1px 2px 0px rgba(0, 0, 0, 0.05);
+  position: relative;
+  flex-wrap: wrap;
+  cursor: ${({ disabled }) => (disabled ? "not-allowed" : "default")};
```
Comment (gm-global) [inline src/components/tagsField/TagsField.styles.ts]: same as above

File: src/components/tagsField/TagsField.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    onClick = () => {},
+    error = false,
+    label,
+    ...rest
+  } = props;
+
+  const renderTags = () => {
+    return value.map((item, index) => (
+      <Badge
+        key={index}
```
Comment (gm-global) [inline src/components/tagsField/TagsField.tsx]:
```suggestion
      <Badge
        key={item.label}
```

File: src/components/tagsField/TagsField.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      />
+    ));
+  };
+
+  return (
+    <StyledWrapper error={error} {...rest}>
+      {label && (
+        <StyledLabel
+          htmlFor={props.id || label}
+          data-testid="label"
```
Comment (gm-global) [inline src/components/tagsField/TagsField.tsx]: can we make this a little unique?

**Review by gm-global — APPROVED** [review]

**Pull request merged**